### PR TITLE
Add RV64GC target

### PR DESCRIPTION
**Pull request by ghost**
RV64GC can use essentially the same set of parameters as RV32IMC, but must be declared in rules_rust for downstream targets to use Rust inside Bazel for RV64GC targets.

Tested against a demonstration project targeting QEMU's RV64 Virt platform.

### CODE REVIEW

File: rust/platform/triple_mappings.bzl [hunk 1]
```diff
@@ -61,6 +62,8 @@ _CPU_ARCH_TO_BUILTIN_PLAT_SUFFIX = {
     "powerpc64le": None,
     "riscv32": "riscv32",
     "riscv32imc": "riscv32",
+    "riscv64": "riscv64",
+    "riscv64gc": "riscv64",
```
Comment (UebelAndre) [inline rust/platform/triple_mappings.bzl:66]: I don't see a constraint for this defined at https://github.com/bazelbuild/platforms/blob/0.0.6/cpu/BUILD. Won't this cause problems if anyone tries to match it?

**Review by UebelAndre — CHANGES_REQUESTED** [review]

File: rust/platform/triple_mappings.bzl (hunk 1 above)
Comment (UebelAndre) [reply rust/platform/triple_mappings.bzl:66]: oh nevermind, they get mapped to `riscv54`. No issue here.

**Review by UebelAndre — APPROVED** [review]
Thanks!

**Comment by UebelAndre** [comment]
A rebase is needed and then the changes can be merged 😄

**Comment by tdewey-rivos** [comment]
Had to re-pin again, syn jumped from .104 to 1.05. Thanks for the review!

**Pull request merged**